### ISSUE TITLE
Ruby 2.2 error with mongoize_time 

### ISSUE THREAD

**Issue by josegrad** [mongoid/mongoid#3958]
Hi,
I have a Rails 4.1.0 app where I just changed to use Ruby 2.2 and gives the error below.
Mongoid version is 4.0.1

```
NoMethodError - undefined method `year' for nil:NilClass:
  activesupport (4.1.0) lib/active_support/values/time_zone.rb:290:in `parse'
   () Users/josegranado/.rvm/gems/ruby-2.2.0@rails4.1.0/bundler/gems/mongoid-35aea60b285a/lib/mongoid/extensions/string.rb:48:in `__mongoize_time__'
   () Users/josegranado/.rvm/gems/ruby-2.2.0@rails4.1.0/bundler/gems/mongoid-35aea60b285a/lib/mongoid/extensions/date.rb:64:in `mongoize'
   () Users/josegranado/.rvm/gems/ruby-2.2.0@rails4.1.0/bundler/gems/mongoid-35aea60b285a/lib/mongoid/fields/standard.rb:10:in `mongoize'
   () Users/josegranado/.rvm/gems/ruby-2.2.0@rails4.1.0/bundler/gems/mongoid-35aea60b285a/lib/mongoid/attributes.rb:291:in `typed_value_for'
   () Users/josegranado/.rvm/gems/ruby-2.2.0@rails4.1.0/bundler/gems/mongoid-35aea60b285a/lib/mongoid/attributes.rb:174:in `block in write_attribute'
   () Users/josegranado/.rvm/gems/ruby-2.2.0@rails4.1.0/bundler/gems/mongoid-35aea60b285a/lib/mongoid/threaded/lifecycle.rb:26:in `_assigning'
   () Users/josegranado/.rvm/gems/ruby-2.2.0@rails4.1.0/bundler/gems/mongoid-35aea60b285a/lib/mongoid/attributes.rb:170:in `write_attribute'
   () Users/josegranado/.rvm/gems/ruby-2.2.0@rails4.1.0/bundler/gems/mongoid-35aea60b285a/lib/mongoid/fields.rb:457:in `block (2 levels) in create_field_setter'
   () Users/josegranado/.rvm/gems/ruby-2.2.0@rails4.1.0/bundler/gems/mongoid-35aea60b285a/lib/mongoid/attributes/processing.rb:97:in `process_attribute'
   () Users/josegranado/.rvm/gems/ruby-2.2.0@rails4.1.0/bundler/gems/mongoid-35aea60b285a/lib/mongoid/attributes/processing.rb:25:in `block in process_attributes'
   () Users/josegranado/.rvm/gems/ruby-2.2.0@rails4.1.0/bundler/gems/mongoid-35aea60b285a/lib/mongoid/attributes/processing.rb:23:in `process_attributes'
   () Users/josegranado/.rvm/gems/ruby-2.2.0@rails4.1.0/bundler/gems/mongoid-35aea60b285a/lib/mongoid/attributes.rb:205:in `block in assign_attributes'
   () Users/josegranado/.rvm/gems/ruby-2.2.0@rails4.1.0/bundler/gems/mongoid-35aea60b285a/lib/mongoid/threaded/lifecycle.rb:26:in `_assigning'
   () Users/josegranado/.rvm/gems/ruby-2.2.0@rails4.1.0/bundler/gems/mongoid-35aea60b285a/lib/mongoid/attributes.rb:204:in `assign_attributes'
   () Users/josegranado/.rvm/gems/ruby-2.2.0@rails4.1.0/bundler/gems/mongoid-35aea60b285a/lib/mongoid/persistable/updatable.rb:51:in `update'
  app/controllers/shipments_controller.rb:110:in `parcels'
  actionpack (4.1.0) lib/action_controller/metal/implicit_render.rb:4:in `send_action'
  actionpack (4.1.0) lib/abstract_controller/base.rb:189:in `process_action'
  actionpack (4.1.0) lib/action_controller/metal/rendering.rb:10:in `process_action'
...
```

The model involved looks like this:

```
class Shipment
  include Mongoid::Document
  include Mongoid::Timestamps
   
  field :pickdate, type: Date
  ...
  ...
end 
```

It was previously working fine with Ruby 2.1.5 and I can come back to that version easily, but wanted check if there's any hints how to solve it. Apparently caused by Ruby 2.2 but the resolution might be on mongoid.

**Comment by arthurnn** [comment]
Not sure if this is a issue on ActiveSupport or Ruby or Mongoid.
can you write a isolated test?
thanks

**Comment by dieuit07** [comment]
I get this error too. 
ruby 2.0.0p598, Rails 4.1.0
update ruby to 2.2.0 still get this error

**Comment by joe1chen** [comment]
I'm also seeing this with Ruby 2.2.2 and Mongoid 3.1.6 with Rails 3.2.19.

**Comment by joe1chen** [comment]
This is a bug in Rails ActiveSupport's Timezone parse function.  This is fixed in latest versions of Rails. (See https://github.com/joe1chen/rails/commit/e88da370f190cabd1e9750c5b3531735950ab415).  Unfortunately still not fixed in Rails 3.2.21, so I had to patch Rails 3 to fix this.

**Comment by durran** [comment]
Thanks @joe1chen
